pkg/workloads/console/runner/integration/integration_test.go

```diff
@@ -120,6 +120,7 @@ func mustCreateRoleBinding(roleBinding rbacv1.RoleBinding) {
 func mustAddSubjectsToRoleBinding(rb rbacv1.RoleBinding, subjects []rbacv1.Subject) {
 	rb.Subjects = subjects
 	err := kubeClient.Update(context.TODO(), &rb)
+	time.Sleep(2 * time.Second)
```

Comment:
So the sleep isn't necessarily needed here but it is useful for preventing flaky tests.

I kept on running into the problem of some tests failing because the local envtest `k8s api server` and `etcd` hadn't gotten around to persisting the kubernetes objects that are being created in the above lines, which meant that tests would then fail when they tried to affirm that the kubernetes objects had been created.

I didn't want to spend much time thinking about an elegant fix to this problem, so I went for the path of least resistance and copied what we've done in other parts of the codebase (i.e. adding `sleeps`) where we'd run into the flaky tests problem.

Reply:
Fair enough if we don't want to tackle this particular fixup in this PR, but in lieu of that it would be great to get a comment in the code stating why we've done this, so that anyone who comes across this in the future doesn't have to dig through PR comments.

Reply:
@benwh I've gone for the option of replacing all `Sleep`s with `Eventually`.